### PR TITLE
Beginners intro to Generative Modeling

### LINKED ISSUE #5007: Beginners intro to Generative Modeling
#### Proposed title of article
Beginners intro to Generative Modeling

#### Proposed article introduction

As we know, deep learning has been the most evolving area in Artificial intelligence since the last decade due to research advancements in this area. However, Generative Modeling is a specific field in Deep Learning that has been a topic of interest for the last few years. In this article, we will be looking forward to the concept of Generative Modeling from scratch and understanding its significance.
Before we dig deeper, I will try to present a basic notion of Generative Modeling for our readers by taking an example. Let us suppose we want to classify an image into a dog or a cat. We can solve this problem simply by feeding the image as an input to a convolutional neural network that can output the image's corresponding category. What if we want this process to be reversed? We describe what we want as an input to the model and get the image as output, and this is generative modeling in its simplest.
In this blog, we will dig deep into the concepts of Generative Modeling and also try to understand how it can impact the future of ML.

#### Key takeaways 

- Introduction to Generative Modeling.
- Generative vs Discriminative Modeling
- Why generative modeling is future of AI.
- GANs

#### Article quality
Most of the articles present on the Internet about Generative Modeling are vague and filled with jargon. This blog clears the basic concepts of Generative Modeling using many examples that a reader can relate to.

#### References
NA

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by tanmay69**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [ ] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [ ] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5007

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by hectorkambow** [comment]
@lalith1403  

Will you be able to finish this one before Monday? 

It may be best to let another PR (peer reviewer) take this one if needed.

**Comment by lalith1403** [comment]
I had finished a preliminary round of review before this one was uploaded. I have gone through the article and can positively complete it.

**Comment by LinusMuema** [comment]
plagiarism check done :heavy_check_mark:

**Comment by tanmay69** [comment]
@lalith1403 Please review this article.

File: content/articles/beginners-intro-to-generative-modeling/index.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+Although I have tried to keep this blog as self-contained as possible, basic prior knowledge of probability and machine learning is still required. Therefore, here is an excellent blog on the conditional probability that readers can refer to before reading the article. 
+
+### Key takeaways
+
+After finishing this blog, readers will understand what a generative model is, how it works, the difference between generative and discriminative models, and how it can impact the future of AI.
+
+### Introduction
+
+Most machine learning practitioners are exposed to the classification or the regression tasks in machine learning first due to its extensive scope and straightforward approach. However, the generative modeling that we are going to cover in this article is less understood. So let's dive deep into the concepts of Generative modeling and understand how it can affect the future of machine learning.
```
[the remark this answers is not in the document]
Comment (lalith1403) [reply content/articles/beginners-intro-to-generative-modeling/index.md:44]: No need to capitalize generative modeling.

File: content/articles/beginners-intro-to-generative-modeling/index.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
... the above paragraph by taking a real-life example. Consider a scenario where you get appointed as a machine learning engineer in some company. Now, you give this excellent news to your parents, and now they ask you what an ML engineer is and what will be your everyday job. Let us take your information about an ML engineer as $x$. To make someone understand who has never studied ML in their life, you have to give some basic information only because too much of the technicalities will be of no use. Now you explain your job to parents in leman terms, let the information they learnt as $z$(hidden representation of actual information $x$). Now using this $z$ as an inspiration, they will draw a picture in their mind of what an ML engineer does, $x^*$. Now ideally, you want $x^*$ to be almost the same as $x$, but it will solely depend on the hidden representation of the $x$. Again, if you have to give the same information to one of your colleagues who knows ML, the $z$ will be different, and the $x^*$ will be closer to the $x$. The same applies to generative modeling. Different models will learn different hidden representations $z$ of original data, and the generated data becomes unique.
```
Comment (lalith1403) [inline content/articles/beginners-intro-to-generative-modeling/index.md:54]: Break this into smaller paragraphs and a sub section if necessary.

File: content/articles/beginners-intro-to-generative-modeling/index.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+### Discriminative vs Generative Modeling
+
+Machine learning models can broadly be categorized into two categories, Discriminative and Generative Models. Before we compare both of them, let's talk about the discriminative models. In simple terms, as its name suggests, the discriminative model aims to discriminate between multiple data instances. It takes input data for training and makes predictions for the unseen data. Most of the classification and regression techniques fall under this category.
```
Comment (lalith1403) [inline content/articles/beginners-intro-to-generative-modeling/index.md:58]: discriminative and generative.

File: content/articles/beginners-intro-to-generative-modeling/index.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+The discriminative model aims to capture $p(Y|X)$, that is, the conditional probability of $Y$ given $X$. In contrast, the generative model aims to capture joint probability $p(X, Y)$, or just $p(X)$ if there is no $Y$ present. 
+
+### Why Generative Modeling?
+
+>*The Future of AI is Generative, not Discriminative.*
+> 
+> Steve Omohundro, Research Scientist at Facebook 
+
+Generative modeling is a most underrepresented topic in machine learning. However, still, a lot of research is going on this topic. This section will discuss why generating data is essential and how it can transform the future of machine learning by taking a few interesting use-cases.
```
Comment (lalith1403) [inline content/articles/beginners-intro-to-generative-modeling/index.md:73]: How are you saying it is underrepresented? When you make such claims, you should back it up with data. If not, it is not ideal to make such claims.

File: content/articles/beginners-intro-to-generative-modeling/index.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+The most trending topic in generative modeling is GAN. It is the most effective technique when it comes to generating data. We will talk more about GAN in the next section. For now, let's stick to its applications. GANs have transformed the world of deep learning and are considered the most remarkable invention in artificial intelligence. From generating portraits to the whole new virtual world, GAN never fails to surprise. [Here](https://www.cnet.com/news/ai-made-portrait-sells-at-christies-auction-for-432500/) is one such exciting story of GANs. In 2018, a French art collective named Obvious used GAN to generate a portrait sold for half a million dollars.
```
Comment (lalith1403) [inline content/articles/beginners-intro-to-generative-modeling/index.md:75]: How do you say it is the most effective technique to generate data. Cite sources?

File: content/articles/beginners-intro-to-generative-modeling/index.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+One more important application of generative modeling is data augmentation. Of course, the most crucial aspect for applying any ML technique is training data. Still, there are certain areas where we have a lot of constraints associated with data availability. One such sector is the medical field. There are a lot of applications of machine learning in the medical field, from diagnosing disease to finding its cure. But medical datasets are harder to collect. Here generative modeling plays a vital role to produce the synthetic data to enlarge the training dataset. This process of extending the training dataset by producing the synthetic data is called Data Augmentation. There is much work done in data augmentation using generative modeling. For example, one search work is done by [Maayan Frid-Adar and others](https://arxiv.org/abs/1803.01229) where they used GAN for synthetic image augmentation to increase the existing performance for liver lesion classification.
```
Comment (lalith1403) [inline content/articles/beginners-intro-to-generative-modeling/index.md:83]: For example, one such work....

File: content/articles/beginners-intro-to-generative-modeling/index.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+### GAN
+
+A generative adversarial network or a GAN is a technique of generative modeling where we train two neural networks simultaneously, one of which is the generator that generates new data. The other one is discriminator, which classifies it as an original sample from a training dataset or data generated by the generator. 
+
+The word __adversarial__ in GAN refers to the competition like the arrangement between generator and discriminator where the generator tries to fool the discriminator. The discriminator's job is to detect the fake data generated by the generator. To understand this, let's imagine a scenario where a thief tries to steal a painting from a museum by replacing it with a fake painting, whereas there will be a curator whose job will be to detect whether the painting is real or fake. In this example, our thief is the generator, and the curator is the discriminator.
```
Comment (lalith1403) [inline content/articles/beginners-intro-to-generative-modeling/index.md:89]: Sentence has grammatical errors.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Hello @tanmay69 
Several changes are required. Also, break paragraphs into shorter paragraphs and ensure that you are looking out for spelling and grammatical errors. 

Tech review to be done.

**Comment by hectorkambow** [comment]
@tanmay69 

Good afternoon and thank you for submitting your topic suggestion. Not a problem but it seems that this PR is still quite a bit aways from being completed. 
 
In an effort to be fair to our larger EngEd community as a rule of thumb - we can only accept content (pull requests) that are complete and fully polished articles/tutorial - meaning they are ready to be published. 
 
All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) - this allow all revisions to be done on your forked personal repo. 
 
We will be closing this PR and please re-submit it when it is ready to be published as a new PR (and you can add the duplicate label) and ping the reviewer and myself. 

>NOTE: A duplicate PR will go back into the queue and will not received any further major revisions. A duplicate PR should be 100% ready to be published, otherwise it will be closed immediately. 

Closing a PR does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.
 
👍 



 cc @lalith1403

**Pull request closed without merging**